src/Forms/GridFieldRefreshButton.php

```diff
@@ -0,0 +1,154 @@
+<?php
+/**
+ * Adds a "Refresh" button to the bottom or top of a GridField.
+ *
+ * @package forms
+ * @subpackage fields-gridfield
+ */
+
+class GridFieldRefreshButton implements GridField_HTMLProvider, GridField_ActionProvider, GridField_URLHandler
+{
+    /**
+     * @var array
+     * @config
+     */
+    private static $allowed_actions = ["check"];
+
+    /**
+     * Fragment to write the button to.
+     * @var string
+     */
+    protected $targetFragment;
+
+    /**
+     * @param string $targetFragment The HTML fragment to write the button into
+     */
+    public function __construct($targetFragment)
+    {
+        $this->targetFragment = $targetFragment;
+    }
+
+    /**
+     * @param GridField $gridField
+     * @return array
+     */
+    public function getHTMLFragments($gridField)
+    {
+        Requirements::javascript('silverstripe-maintenance/javascript/CheckForUpdates.js');
+
+        $button = GridField_FormAction::create(
+            $gridField,
+            'refresh',
+            _t('GridFieldRefreshButton.REFRESH', 'Check for updates'),
+            'refresh',
+            null
+        );
+
+        $button->setAttribute('data-icon', 'arrow-circle-double');
+        $button->setAttribute('data-check', $gridField->Link('check'));
+        $button->setAttribute(
+            'data-message',
+            _t(
+                'GridFieldRefreshButton.MESSAGE',
+                'Updating this list may take 2-3 minutes. You can continue to use the CMS while we run the update.'
+            )
+        );
+
+        if ($this->hasActiveJob()) {
+            $button->setTitle(_t('GridFieldRefreshButton.UPDATE', 'Updating...'));
+            $button->setDisabled(true);
+        }
+
+        return [
+            $this->targetFragment => '<p class="grid-refresh-button" id="checkForUpdates">' . $button->Field() . '</p>',
```

Comment:
Can you please use a template to render this HTML rather than doing it in PHP? `$this->targetFragment = ArrayData::create(['Field' => $button->Field()])->renderWith(__CLASS__)` for example